### PR TITLE
Add initial IPv6 documentation

### PR DESCRIPTION
**Pull request by johngmyers**

### CODE REVIEW

File: docs/networking/ipv6.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+# CNI
+
+kOps currently supports IPv6 on Calico, Cilium, and bring-your-own CNI only.
+
+CNIs must not masquerade IPv6 addresses.
+
+## Calico
+
+Running IPv6 with Calico requires a Debian 11-based AMI. As of the writing of this document, Ubuntu does not work due to an 
+[issue with systemd's handling of AWS's incorrect DHCP responses](https://bugs.launchpad.net/ubuntu/+source/systemd/+bug/1945888).
```
Comment (johngmyers) [inline docs/networking/ipv6.md:31]: Do we have a public ticket reference for the AWS DHCP bug?  Do we want to describe it in the "Cloud providers" section?

Comment (hakman) [reply docs/networking/ipv6.md:31]: Not that I know of.

File: docs/networking/ipv6.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+## CNI
+
+kOps currently supports IPv6 on Calico, Cilium, and bring-your-own CNI only.
+
+CNIs must not masquerade IPv6 addresses.
+
+### Calico
+
+Running IPv6 with Calico requires a Debian 11-based AMI. As of the writing of this document, Ubuntu does not work due to an 
+[issue with systemd's handling of AWS's incorrect DHCP responses](https://bugs.launchpad.net/ubuntu/+source/systemd/+bug/1945888).
```
Comment (hakman) [inline docs/networking/ipv6.md:31]: https://github.com/systemd/systemd/issues/20803 may be a better reference.

Comment (johngmyers) [reply docs/networking/ipv6.md:31]: I was trying to go for something Ubuntu-specific, but I suppose AL2 would probably have the same issue.

Comment (hakman) [reply docs/networking/ipv6.md:31]: I don't know if AL2 uses systemd-networkd. Each distro uses something different. I Just mention Debian 11 if it works with Cilium, and mention that Ubuntu 22.04 should work too.

Comment (johngmyers) [reply docs/networking/ipv6.md]: Perhaps you could refine the language after the PR lands? I don't have complete enough knowledge of all the possibilities.

File: docs/networking/ipv6.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
+# IPv6
+
+{{ kops_feature_table(kops_added_ff='1.22') }}
```
Comment (hakman) [inline docs/networking/ipv6.md:3]: Best not to confuse people:
```suggestion
{{ kops_feature_table(kops_added_ff='1.23') }}
```

File: docs/networking/ipv6.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+kOps has experimental support for configuring clusters with IPv6-only pods and dual-stack nodes.
```
Comment (hakman) [inline docs/networking/ipv6.md:5]: Dual-stack should probably not get mentioned.
```suggestion
kOps has experimental support for configuring clusters with IPv6-only pods on AWS instances with IPv6 enabled.
```

**Review by hakman — CHANGES_REQUESTED** [review]

File: docs/networking/ipv6.md (hunk 2 above)
Comment (hakman) [reply docs/networking/ipv6.md]: 👍

File: docs/networking/ipv6.md (hunk 3 above)
Comment (johngmyers) [reply docs/networking/ipv6.md:3]: We did feature flag it in 1.22

File: docs/networking/ipv6.md (hunk 4 above)
Comment (johngmyers) [reply docs/networking/ipv6.md:5]: The nodes are dual-stack.

File: docs/networking/ipv6.md (hunk 3 above)
Comment (hakman) [reply docs/networking/ipv6.md:3]: Yes, but it's a totally different approach that we don't want to document or support. Mentioning 1.22 would confuse some users in thinking that it's same as what we will release in 1.23. Am I wrong?

Comment (johngmyers) [reply docs/networking/ipv6.md:3]: I suppose it depends on what the table is supposed to mean and how people interpret "experimental". I suspect there's a good chance we'll remove the feature flag for 1.23.

Comment (hakman) [reply docs/networking/ipv6.md:3]: I agree that the feature flag will probably be removed in 1.23. Though, kOps docs don't have a way to say, experimental in version X, beta in version Y, stable in version Z. I am afraid people will try this now with 1.22 and get something different than described here.

**Review by hakman — APPROVED** [review]

**Pull request merged**